Add SQA design documents for NS and ray tracing

GiudGiud opened the pull request:
@grmnptr @lindsayad @tanoret fyi

I think we can expand on what we know we are missing at this point

My limited knowledge of the work on the FE side is likely transparent in this PR.

GiudGiud committed:
Add missing design documents for the Navier Stokes module, refs #23049

GiudGiud committed:
Exchange scope and purpose in FSI module refs #23049

GiudGiud committed:
Add missing document design for the Ray Tracing module, refs #23049

lindsayad commented on modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md:
```diff
@@ -1 +1,49 @@
+!template! item key=system-purpose
+!! system-purpose-begin
+The purpose of this software is to allow simulation of fluid flow in regular and porous media.
+It should be able to determine the pressure field and velocity fields in the fluid domain, as well
+as the temperature field in the case of conjugate heat transfer simulations. It aims to also be able to
```
```suggestion
It should be able to determine the pressure, velocity, and fluid temperature fields, as well
as the solid temperature field in the case of conjugate heat transfer simulations. It aims to also be able to
```

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1305508/) on 62f5d45 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/23195/site)

This comment will be updated on new commits.

lindsayad commented on modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md:
```diff
@@ -1 +1,49 @@
+as the temperature field in the case of conjugate heat transfer simulations. It aims to also be able to
+transport scalar species in the flow fields determined.
+
+This software is intended as a coarse mesh computational fluid dynamics simulation tool and to be the
+numerical backbone of the Pronghorn software. Its goals include the simulation of coolant flow in
```
I might delete this sentence. I think it is unnecessarily narrowing. In the very short term the module will also be the backbone of the SAM multi-D capability. Moreover, though the scope for NEAMS is coarse mesh, I am hopeful that if not now, then some day it can also be used for finer meshes as well, e.g. when we have scalable solvers

GiudGiud replied:
We could update when the SAM multi-D is merged in.
Similarly once we have shown we can do finer meshes in some capacity we can update.

but actually are these documents updateable when needed @cticenhour ?

cticenhour replied:
These documents are living ones, meant to be kept up to date as the code they're associated with change. I tend toward more broad statements and refer back to design documentation for specifics myself, but I leave those sorts of things to the module maintainers for how they want to outline the contents of their code.

GiudGiud replied:
removed. We ll update after SAM merge

lindsayad reviewed (dismissed):
Just reviewing the NS part. Very nice

GiudGiud committed:
Address Alex's review of NS docs
- dont mention Pronghorn as the main purpose as things are still in flux with SAM
- Update modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_srs.md

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>

Make some links optional to avoid breakage for apps that only pull some modules in

Update modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_sdd.md

Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>

GiudGiud commented:
Review should be addressed

Still need eyes on ray tracing. Logan and I wrote it together so need another reviewer

cticenhour commented on modules/navier_stokes/doc/content/modules/navier_stokes/sqa/navier_stokes_sdd.md:
```diff
@@ -1 +1,89 @@
+The module is designed to be able to handle both, with utilities to perform conversions from one to the
+other as appropriate. The module is not currently designed to handle transitions between the two variable sets at a fluid interface.
+
+Each major class of discretizations has its own index page, notable the [continuous Galerkin finite element](navier_stokes/cgfe.md),
```
```suggestion
Each major class of discretizations has its own index page, notably the [continuous Galerkin finite element](navier_stokes/cgfe.md),
```

cticenhour reviewed (changes requested):
Only one typo I found. Anymore work needed here @GiudGiud?

GiudGiud commented:
nope, good to go as far as I know. Typo fix is squashed in

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1321771/) on 5ed8c31 : invalidated by @GiudGiud

fluke
